solutions/usaco-738.mdx

```diff
@@ -2,19 +2,68 @@
 id: usaco-738
 source: USACO Silver 2017 US Open
 title: Paired Up
-author: Sofia Yang
+author: Sofia Yang, Ryan Chou
 
 ---
 
-[Official Analysis](http://www.usaco.org/current/data/sol_pairup_silver_open17.html)
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_pairup_silver_open17.html)
 
+Let's say we have three distinct values in our dataset, 
+
+$A < B < C$
+
+In this case, pairing up $B + C$ would always be greater than $A + C$.
+
+Therefore, we should pair up the greatest value ($C$), with the smallest value ($A$), in order to achieve the most optimal milking time.
+
+## Implementation
 **Time Complexity:** $\mathcal{O}(N\log N)$
 
 (Because it requires sorting the input, which is $\mathcal{O}(N\log N)$). 
 
 
 <LanguageSection>
 
+<PySection>
+```py
+import sys
+sys.stdin = open("pairup.in", 'r')
+sys.stdout = open("pairup.out", 'w')
+
+n = int(input())
+all_cows = []
+
+for _ in range(n):
+    num_cows, milk_time = map(int, input().split())
+    all_cows.append([milk_time, num_cows])
+
+# sort by the amount of time the cows take to milk.
+all_cows.sort()
+
+left, right = 0, n - 1
+
+ans = 0 
+
+while left <= right:
+    # how many cows have been grouped together.
+    sub = min(all_cows[left][1], all_cows[right][1])
+    """
+    since we have to take account of the total time
+    to milk all cows, we're getting the maximum time
+    out of all the milkings.
+    """
```

Comment:
not necessary
just feels a bit obvious to anyone who's even skimmed the problem

Reply:
wait why was this resolved

Reply:
i removed it